lib/src/_shared/hooks/usePickerState.ts

```diff
@@ -89,24 +88,19 @@ export function usePickerState<TInput, TDateValue>(
 
         setMobileKeyboardViewOpen(!isMobileKeyboardViewOpen);
       },
-      onDateChange: (
-        newDate: TDateValue,
-        currentVariant: WrapperVariant,
-        isFinish: boolean | symbol = true
-      ) => {
+      onDateChange: (newDate: TDateValue, selectionState: PickerSelectionState = 'partial') => {
         setPickerDate(newDate);
-        const isFinishing =
-          typeof isFinish === 'boolean' ? isFinish : isFinish === FORCE_FINISH_PICKER;
-
-        if (isFinishing) {
-          const autoAcceptRequested = Boolean(autoOk) || isFinish === FORCE_FINISH_PICKER;
-          if (currentVariant === 'mobile' && autoAcceptRequested) {
-            acceptDate(newDate, true);
-          }
-
-          if (currentVariant !== 'mobile') {
-            acceptDate(newDate, autoAcceptRequested);
-          }
+
+        if (selectionState === 'partial') {
+          acceptDate(newDate, false);
+        }
+
+        if (selectionState == 'finish') {
```

Comment:
It seems that we miss an eslint rule here:

```suggestion
        if (selectionState === 'finish') {
```

It will be solved with #1930